### PR TITLE
[systemtest] ConnectS2IST: fix tests

### PR DESCRIPTION
**Pull request by im-konge**
Signed-off-by: Lukas Kral <lkral@redhat.com>

### Type of change
- Bugfix

### Description

This PR fix some of ConnectS2IST tests:
```
testJvmAndResources - added dynamic wait for resource deletion
testDeployS2IWithMongoDBPlugin - remove wait for log and add wait for ready state in status
```

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

**Comment by im-konge** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectS2IST#testJvmAndResources,testDeployS2IWithMongoDBPlugin test_only

**Comment by im-konge** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectS2IST#testJvmAndResources+testDeployS2IWithMongoDBPlugin test_only

**Comment by im-konge** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectS2IST#testJvmAndResources+testDeployS2IWithMongoDBPlugin test_only

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectS2IST test_only

**Comment by im-konge** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectS2IST test_only

**Comment by im-konge** [comment]
All tests are green on my machine, I tested it 5 times. So from my POV ready for review.

File: systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java [hunk 1]
```diff
@@ -89,17 +90,19 @@ void testDeployS2IWithMongoDBPlugin() throws InterruptedException, IOException {
...
+            LOGGER.info("Create Connector result: {}", createConnectorOutput);
+            return !createConnectorOutput.contains("error_code");
+        });
 
         // Make sure that connector is really created
         Thread.sleep(10_000);
 
         String connectorStatus = cmdKubeClient().execInPod(kafkaClientsPodName, "curl", "-X", "GET", "http://" + KafkaConnectS2IResources.serviceName(kafkaConnectS2IName) + ":8083/connectors/" + kafkaConnectS2IName + "/status").out();
+
+        KafkaConnectS2IUtils.waitForConnectS2IStatus(kafkaConnectS2IName, "Ready");
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java]: Isn't this line useless ?

Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java]: Removed :)

**Pull request merged**